controllers/ingressnodefirewallconfig_controller.go

```diff
@@ -86,9 +90,30 @@ func (r *IngressNodeFirewallConfigReconciler) Reconcile(ctx context.Context, req
 	}
 
 	if err := r.syncIngressNodeFwConfigResources(ctx, instance); err != nil {
-		return ctrl.Result{}, err
+		condition = status.ConditionDegraded
+	} else {
+		err = status.IsIngressNodeFirewallConfigAvailable(ctx, r.Client, req.NamespacedName.Namespace)
+		if err != nil {
+			if _, ok := err.(status.IngressNodeFirewallConfigResourcesNotReadyError); ok {
+				ctrResult = ctrl.Result{RequeueAfter: 5 * time.Second}
+			}
+			condition = status.ConditionProgressing
+		} else {
+			condition = status.ConditionAvailable
+		}
+	}
+
+	if condition != "" {
+		if err != nil {
+			if errors.Unwrap(err) != nil {
+				wrappedErrMsg = errors.Unwrap(err).Error()
+			}
+		}
+		if err := status.Update(context.TODO(), r.Client, instance, condition, errorMsg, wrappedErrMsg); err != nil {
```

Comment:
The same here. If that err is to be reported / returned in 116, it should IMO be err = status.Update ? On the other hand, I imagine that you only want to log this one here, and not return it in the reconciler

Reply:
yeah sorry didn't mean to initalize new one :(